EGW NAT Stats Troubleshooting & EGW Docs Structure Improvements

tommyp1ckles opened the pull request:
Provides separate page that delves into topic of troubleshooting issues of EGW SNAT map overflow, including how to monitor this with the new nat-stats metrics and mitigation strategies.

Because this is a deeper topic, this PR seeks to add this as an external page to avoid having too much esoteric information on the general EGW docs page.

As well, fixes bad inline code section in metrics from previous docs changes related to nat stats metrics.

tommyp1ckles committed:
Documentation: add page for egress gateway troubleshooting.

New document is a general troubleshooting document for egress-gateway.
This probably belongs in its own page as the content is more detailed
then the generate egress-gateway documentation page (which has a troubleshooting
page around fixing common issues with EGW CEGP configuration issues.

This also references this document at the bottom of the egress-gateway
page.

Signed-off-by: Tom Hadlaw <tom.hadlaw@isovalent.com>

tommyp1ckles committed:
Documentation/metrics.rst: fix unnecessary back-quoting on metric name.

Should just be one back quote instead of three, this was being rendered
to show one the inner quotes when viewing the HTML documentation.

Signed-off-by: Tom Hadlaw <tom.hadlaw@isovalent.com>

derailed reviewed (approved):
@tommyp1ckles Nice work Tom!

tommyp1ckles commented on Documentation/network/egress-gateway.rst:
```diff
@@ -481,3 +481,12 @@ will be the ``egressIP`` from the policy, if specified).
 policy, check that the pod(s) and egress node are labeled correctly to match
 the policy selectors.
+
+Troubleshooting Guide
```
Good point.  I want to keep this topic in a seperate page from the general EGW page, as it feels pretty esoteric for the general EGW guide.

I did a direct link instead of the TOC. As well, I added the separate troubleshooting page as EGW "Advanced Troubleshooting" since the linter complains if there are pages not included in the root level TOC:

![image](https://github.com/cilium/cilium/assets/5881740/a038ef6d-6f00-4691-99e9-202617010774)

tommyp1ckles replied:
I do wonder if having EGW under "External networking" even makes sense.  EGW is a feature in itself - it may make sense to split the EGW articles into their own dedicated section:

i.e. the following structured

* External networking
    * Setting up Support for External Workloads (beta)
* Egress Gateway
    * Getting started
    * Advanced Troubleshooting

@julianwiedmann wdyt?

tommyp1ckles replied:
Actually, I'm sort of leaning towards moving to a dedicated EGW section, see:

Index Page:

![image](https://github.com/cilium/cilium/assets/5881740/e3f8a6e5-4e5b-47b2-bfd7-661db6240f7d)

And the new EGW sub-index:

![image](https://github.com/cilium/cilium/assets/5881740/e6a5500e-f0df-48e0-9189-c313c1859959)

networkop replied:
this makes a lot of sense. these two should never have been grouped together

tommyp1ckles committed:
Documentation: create dedicated egress-gateway section on index.

EGW is a feature on it's own. With the addition of the advanced
troubleshooting topics the "external networking" section seems to no
longer adequately encapsulate it's various topics.

This splits EGW and Ext Networking sections using the following
structure.

    External networking
        Setting up Support for External Workloads (beta)
    Egress Gateway
       	Egress Gateway
        Advanced Troubleshooting

Signed-off-by: Tom Hadlaw <tom.hadlaw@isovalent.com>

tommyp1ckles committed:
Documentation: move egress gateway pages to separate dir.

Now only egw rst file in top level Documentation/networking is
the TOC file.
All further EGW specific docs will be put in: Documentation/networking/egress-gateway.

This will also make managing CODEOWNER ownership easier, subsequent
commits will move ownership to new directory.

Signed-off-by: Tom Hadlaw <tom.hadlaw@isovalent.com>

tommyp1ckles committed:
CODEOWNERS: Update ownership for EGW docs dir restructure.

Now egw/docs-structure teams will be called to review:

* EGW TOC page in ./networking
* All child pages in ./networking/egress-gateway.

This will make it easier to manage ownership in the future
as all new EGW pages in the egw dir will be owned by
egw & docs-structure teams.

Signed-off-by: Tom Hadlaw <tom.hadlaw@isovalent.com>

bimmlerd reviewed (approved):
CODEOWNERS lgtm

julianwiedmann reviewed (approved):
looks good, thank you! I labeled for backport to 1.16, as that's where we introduced the NAT stats feature.

networkop commented on Documentation/network/egress-gateway/egress-gateway-troubleshooting.rst:
```diff
@@ -0,0 +1,60 @@
+SNAT Connection Limits
+----------------------
+
+In use-cases where egress-gateway is being used to masquerade traffic to a small set of remote endpoints, it's possible
+to cause issues by exceeding the max number of source IPs that can be allocated by Cilium's NAT mapping per remote endpoint.
```
is it possible add a little explanation why this happens (specifically why small set of remote endpoints)?

networkop replied:
nvm, should've read a bit more before commenting
